src/benchmarklib/tpch/tpch_db_generator.cpp

```diff
@@ -191,21 +196,31 @@ TpchDbGenerator::TpchDbGenerator(float scale_factor, uint32_t chunk_size)
     : _scale_factor(scale_factor), _chunk_size(chunk_size) {}
 
 std::unordered_map<TpchTable, std::shared_ptr<Table>> TpchDbGenerator::generate() {
-  TableBuilder customer_builder{_chunk_size, customer_column_types, customer_column_names, UseMvcc::Yes};
-  TableBuilder order_builder{_chunk_size, order_column_types, order_column_names, UseMvcc::Yes};
-  TableBuilder lineitem_builder{_chunk_size, lineitem_column_types, lineitem_column_names, UseMvcc::Yes};
-  TableBuilder part_builder{_chunk_size, part_column_types, part_column_names, UseMvcc::Yes};
-  TableBuilder partsupp_builder{_chunk_size, partsupp_column_types, partsupp_column_names, UseMvcc::Yes};
-  TableBuilder supplier_builder{_chunk_size, supplier_column_types, supplier_column_names, UseMvcc::Yes};
-  TableBuilder nation_builder{_chunk_size, nation_column_types, nation_column_names, UseMvcc::Yes};
-  TableBuilder region_builder{_chunk_size, region_column_types, region_column_names, UseMvcc::Yes};
+  const auto customer_count = static_cast<size_t>(tdefs[CUST].base * _scale_factor);
+  const auto order_count = static_cast<size_t>(tdefs[ORDER].base * _scale_factor);
+  const auto part_count = static_cast<size_t>(tdefs[PART].base * _scale_factor);
+  const auto supplier_count = static_cast<size_t>(tdefs[SUPP].base * _scale_factor);
+  const auto nation_count = static_cast<size_t>(tdefs[NATION].base);
+  const auto region_count = static_cast<size_t>(tdefs[REGION].base);
+
+  TableBuilder customer_builder{_chunk_size, customer_column_types, customer_column_names, UseMvcc::Yes,
+                                customer_count};
+  TableBuilder order_builder{_chunk_size, order_column_types, order_column_names, UseMvcc::Yes, order_count};
+  TableBuilder lineitem_builder{_chunk_size, lineitem_column_types, lineitem_column_names, UseMvcc::Yes,
+                                order_count * 4};
```

Comment:
A comment/constant for the 4 might be helpful in the future.

Reply:
Same few lines down, not sure it's the same constant.

Reply:
Not sure how to call that constant. It is a number by which the row count is multiplied and is defined as such in the TPC-H. I added a comment explaining that.

Reply:
I think that's good enough 👍